File: .skills/openclaw-skills/skills/godiao/minimax-mcp-tool/scripts/minimax_mcp.js
/**
 * MiniMax Web Search + Image Understanding via MCP Server
 * Usage:
 *   node minimax_websearch.js search "关键词"
 *   node minimax_websearch.js image "分析要求" "图片路径或URL"
 *
 * Required env vars:
 *   MINIMAX_API_KEY    - MiniMax API key (from MiniMax Token Plan)
 *   MINIMAX_PYTHON     - Path to Python with minimax-coding-plan-mcp installed
 *                        (default: E:\.uv-venv\Scripts\python.exe)
 *
 * Example openclaw.json config:
 *   { "env": { "MINIMAX_API_KEY": "...", "MINIMAX_PYTHON": "E:\\.uv-venv\\Scripts\\python.exe" } }
 */
const { execSync } = require('child_process');
const path = require('path');

const VENV_CERTIFI_CA = process.platform === 'win32'
  ? 'E:\\.uv-venv\\Lib\\site-packages\\certifi\\cacert.pem'
  : path.join(process.env.HOME || '/usr/local', '.uv-venv', 'lib', 'certifi', 'cacert.pem');

const apiKey = process.env.MINIMAX_API_KEY;
if (!apiKey) {
  console.error('Error: MINIMAX_API_KEY environment variable is not set.');
  console.error('Add it to openclaw.json: { "env": { "MINIMAX_API_KEY": "your-key" } }');
  process.exit(1);
}

const defaultPy = process.platform === 'win32'
  ? 'E:\\.uv-venv\\Scripts\\python.exe'
  : path.join(process.env.HOME || '/usr/local', '.uv-venv', 'bin', 'python');

const pythonExe = process.env.MINIMAX_PYTHON || defaultPy;
const apiHost = process.env.MINIMAX_API_HOST || 'https://api.minimaxi.com';

function runMCP(messages) {
  const input = messages.map(m => JSON.stringify(m)).join('\n') + '\n';
  const result = execSync(`"${pythonExe}" -m minimax_mcp.server`, {
    env: { ...process.env, MINIMAX_API_KEY: apiKey, MINIMAX_API_HOST: apiHost, FASTMCP_LOG_LEVEL: 'ERROR', REQUESTS_CA_BUNDLE: VENV_CERTIFI_CA },
    input,
    maxBuffer: 10 * 1024 * 1024,
    timeout: 60000,
    shell: true,
    windowsHide: true
  });
  const output = result.toString().trim();
  const lines = output.split('\n');
  const results = [];
  for (const line of lines) {
    const trimmed = line.trim();
    if (!trimmed || !trimmed.startsWith('{')) continue;
    try { results.push(JSON.parse(trimmed)); } catch {}
  }
  return results;
}

const cmd = process.argv[2];
const arg1 = process.argv[3];
const arg2 = process.argv[4];

if (cmd === 'search') {
  if (!arg1) {
    console.error('Usage: node minimax_websearch.js search "关键词"');
    process.exit(1);
  }
  const results = runMCP([
    { jsonrpc: '2.0', id: 1, method: 'initialize', params: { protocolVersion: '2024-11-05', capabilities: {}, clientInfo: { name: 'websearch', version: '1.0' } } },
    { jsonrpc: '2.0', id: 3, method: 'tools/call', params: { name: 'web_search', arguments: { query: arg1 } } }
  ]);
  const sr = results.find(r => r.id === 3);
  if (!sr || sr.error) { console.error('Search error:', sr?.error || 'no response'); process.exit(1); }
  const data = sr.result;
  if (data.isError) { console.error('Tool error:', data.content?.[0]?.text || 'unknown'); process.exit(1); }
  try {
    const parsed = JSON.parse(data.content[0].text);
    const out = { query: arg1, count: parsed.organic?.length || 0, results: parsed.organic || [], related: parsed.related_searches || [] };
    console.log(JSON.stringify(out, null, 2));
  } catch { console.log(data.content[0].text); }

} else if (cmd === 'image') {
  if (!arg1 || !arg2) {
    console.error('Usage: node minimax_websearch.js image "分析要求" "图片路径或URL"');
    process.exit(1);
  }
  const results = runMCP([
    { jsonrpc: '2.0', id: 1, method: 'initialize', params: { protocolVersion: '2024-11-05', capabilities: {}, clientInfo: { name: 'websearch', version: '1.0' } } },
    { jsonrpc: '2.0', id: 4, method: 'tools/call', params: { name: 'understand_image', arguments: { prompt: arg1, image_source: arg2 } } }
  ]);
  const ir = results.find(r => r.id === 4);
  if (!ir || ir.error) { console.error('Image error:', ir?.error || 'no response'); process.exit(1); }
  const data = ir.result;
  if (data.isError) { console.error('Tool error:', data.content?.[0]?.text || 'unknown'); process.exit(1); }
  console.log(data.content[0].text);

} else if (cmd === 'tools') {
  const results = runMCP([
    { jsonrpc: '2.0', id: 1, method: 'initialize', params: { protocolVersion: '2024-11-05', capabilities: {}, clientInfo: { name: 'websearch', version: '1.0' } } },
    { jsonrpc: '2.0', id: 2, method: 'tools/list', params: {} }
  ]);
  const tr = results.find(r => r.id === 2 && r.result?.tools);
  if (!tr) { console.error('Could not get tools list'); process.exit(1); }
  console.log('Available tools:');
  tr.result.tools.forEach(t => console.log(' -', t.name, '-', t.description?.split('\n')[0] || ''));

} else {
  console.error('Usage:');
  console.error('  node minimax_websearch.js search "关键词"');
  console.error('  node minimax_websearch.js image "分析要求" "图片路径或URL"');
  console.error('  node minimax_websearch.js tools');
  process.exit(1);
}

File: .skills/openclaw-skills/skills/hongjiahao371-pixel/minimax-mcp-docker/minimax_mcp.py
#!/usr/bin/env python3
"""
MiniMax MCP Tool - Image Understanding & Web Search
API key is read from environment or credentials file - never hardcoded
"""

import subprocess
import os
import json
import sys

MCP_PATH = "node_modules/@ameno/pi-minimax-mcp/bin/pi-minimax-mcp.js"
DEFAULT_API_HOST = "https://api.minimaxi.com"

def get_api_key():
    """Read API key from environment or credentials file"""
    
    # Try environment variable first
    api_key = os.environ.get('MINIMAX_API_KEY')
    if api_key:
        return api_key
    
    # Try credentials file
    creds_path = os.path.expanduser('~/.openclaw/credentials/minimax.json')
    try:
        with open(creds_path) as f:
            creds = json.load(f)
            return creds.get('api_key')
    except:
        pass
    
    return None

def run_mcp_command(command, args):
    """Run MiniMax MCP command"""
    
    api_key = get_api_key()
    if not api_key:
        return "Error: No API key found. Set MINIMAX_API_KEY or create ~/.openclaw/credentials/minimax.json"
    
    # Build command
    cmd = ['node', MCP_PATH, command] + args
    
    # Set environment
    env = os.environ.copy()
    env['MINIMAX_API_KEY'] = api_key
    env['MINIMAX_API_HOST'] = DEFAULT_API_HOST
    
    # Run
    try:
        result = subprocess.run(
            cmd,
            capture_output=True,
            text=True,
            env=env,
            timeout=120
        )
        
        if result.returncode != 0:
            return f"Error: {result.stderr}"
        
        return result.stdout
        
    except subprocess.TimeoutOut:
        return "Error: Command timed out"
    except Exception as e:
        return f"Error: {str(e)}"

def understand_image(image_path):
    """Understand an image"""
    if not os.path.exists(image_path):
        return f"Error: File not found: {image_path}"
    
    return run_mcp_command('understand', [image_path])

def search_web(query):
    """Search the web"""
    return run_mcp_command('search', [query])

if __name__ == "__main__":
    if len(sys.argv) < 3:
        print("Usage:")
        print("  minimax_mcp.py understand <image_path>")
        print("  minimax_mcp.py search <query>")
        sys.exit(1)
    
    command = sys.argv[1]
    
    if command == "understand":
        if len(sys.argv) < 3:
            print("Error: Missing image path")
            sys.exit(1)
        image_path = sys.argv[2]
        print(understand_image(image_path))
    
    elif command == "search":
        if len(sys.argv) < 3:
            print("Error: Missing search query")
            sys.exit(1)
        query = sys.argv[2]
        print(search_web(query))
    
    else:
        print(f"Unknown command: {command}")
        sys.exit(1)
